### DISCUSSION TITLE
Ability to have editor plugins in lazer.

### DISCUSSION THREAD

**Discussion by Aprixia** [ppy/osu#17505] [Ideas] [answered]
This is a suggestion that would be useful for mappers.

I think it would be cool to have the ability to add custom made plugins to the lazer editor, made by the community. These plugins would add new features and change how some features work to make it easier for the mapper to make good maps.
These can for example be as small as more options for grid snap, checking distance inconsistencies while mapping, notifying about bad overlaps, ...
And larger new features like blanket snap, where the plugin will try to snap the object to blanket with the slider they want to blanket it with. And even changing the layout of the editor. And making aimod more advanced.

These plugins would be made by the community, meaning the dev team would only have to add the support for it, and could for example have a listing page, like I've heard custom gamemodes will have.

I hope this gets added, as it'd be really cool to have and also would be really useful for mappers.

**Comment by peppy** [comment, marked as answer]
We don't have any intentions on adding editor plugin support. But as osu! is open source, developers are welcome to help develop the editor and add features which are required.

**Comment by GopoBoi** [comment]
this has been suggested several times, and the answer has always been that it would be an unreasonable amount of work to just make plugins work and that developers can integrate their features directly into the osu codebase if they want to see it supported in the editor.

**Reply by uzervlad** [reply]
I find it hard to believe that something like "blanket snap" would be accepted into osu!, and there are possibly many other small QoL features that just don't feel right other than as plugins

**Reply by peppy** [reply]
Please open proposals for anything that you current use external apps for and want to see in osu!.
